### PR TITLE
[Py-Client] Support table model with python client

### PR DESCRIPTION
**Pull request by HTHou**
## Description

This PR implements the session Api for table model.
The usage can be find in TableModelSessionExample.py and TableModelSessionPoolExample.py.

### CODE REVIEW

File: iotdb-client/client-py/TableModelSessionPoolExample.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        print(res.next())
+
+    session_pool.put_back(session)
+
+
+def delete_data():
+    session = session_pool.get_session()
+    session.execute_non_query_statement("drop database db1")
+    print("data has been deleted. now the devices are:")
+    res = session.execute_statement("show databases")
```
Comment (jt2594838) [inline iotdb-client/client-py/TableModelSessionPoolExample.py:120]: devices -> databases

File: iotdb-client/client-py/TableModelSessionPoolExample.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    time_zone="UTC+8",
+    max_retry=3,
+    sql_dialect="table",
+    database="db1",
+)
+max_pool_size = 5
+wait_timeout_in_ms = 3000
+
+# Create a session pool
+session_pool = SessionPool(pool_config, max_pool_size, wait_timeout_in_ms)
```
Comment (jt2594838) [inline iotdb-client/client-py/TableModelSessionPoolExample.py:144]: Just curious: what is the standard of putting configs in `pool_config` or not? ` max_pool_size` and `wait_timeout_in_ms` seem perfectly fit in `pool_config` but they are listed separately.

File: iotdb-client/client-py/iotdb/Session.py [hunk 3]
```diff
@@ -1058,6 +1076,35 @@ def insert_aligned_tablets(self, tablet_lst):
...
+    def insert_relational_tablet(self, tablet):
+        """
+        insert one tablet, in a tablet, for each timestamp, the number of measurements is same
+            for example three records in the same device can form a tablet:
+                timestamps,     m1,    m2,     m3
+                         1,  125.3,  True,  text1
+                         2,  111.6, False,  text2
+                         3,  688.6,  True,  text3
+        :param tablet: a tablet specified above
+        """
```
Comment (jt2594838) [inline iotdb-client/client-py/iotdb/Session.py:1088]: Better to explicitly indicate how devices are expressed in a relational Tablet (i.e., ID columns) in the example, which is quite different from the tree Tablet.

File: iotdb-client/client-py/iotdb/Session.py [hunk 4]
```diff
@@ -1058,6 +1076,35 @@ def insert_aligned_tablets(self, tablet_lst):
...
+        request = self.gen_insert_relational_tablet_req(tablet)
+        try:
+            connection = self.get_connection(tablet.get_device_id())
```
Comment (jt2594838) [inline iotdb-client/client-py/iotdb/Session.py:1091]: Better to add some comments to explain that it is the table name here actually, or change the method name to a more generic one.

File: iotdb-client/client-py/iotdb/utils/Field.py [hunk 5]
```diff
@@ -38,13 +38,19 @@ def copy(field):
...
                 output.set_float_value(field.get_float_value())
             elif output.get_data_type() == TSDataType.DOUBLE:
                 output.set_double_value(field.get_double_value())
-            elif output.get_data_type() == TSDataType.TEXT:
+            elif (
+                output.get_data_type() == TSDataType.TEXT
+                or output.get_data_type() == TSDataType.STRING
+            ):
                 output.set_binary_value(field.get_binary_value())
             else:
```
Comment (jt2594838) [inline iotdb-client/client-py/iotdb/utils/Field.py:55]: Should DATE and BLOB be processed here?

**Review by jt2594838 — APPROVED** [review]

File: iotdb-client/client-py/iotdb/utils/Field.py (hunk 5 above)
Comment (HTHou) [reply iotdb-client/client-py/iotdb/utils/Field.py:55]: This pr is focus on the table model, I think we need another pr to completely support the new data types.

File: iotdb-client/client-py/TableModelSessionPoolExample.py (hunk 2 above)
Comment (HTHou) [reply iotdb-client/client-py/TableModelSessionPoolExample.py:144]: Don’t know +1. @ycycse do you have any idea?

Comment (ycycse) [reply iotdb-client/client-py/TableModelSessionPoolExample.py:144]: > Don’t know +1. @ycycse do you have any idea?

Configs in `pool_config` is used to construct and connect session which contains relating infos about session. I think change it to `session_config` would be more clear.

File: iotdb-client/client-py/TableModelSessionPoolExample.py (hunk 1 above)
Comment (HTHou) [reply iotdb-client/client-py/TableModelSessionPoolExample.py]: fixed

File: iotdb-client/client-py/iotdb/Session.py (hunk 3 above)
Comment (HTHou) [reply iotdb-client/client-py/iotdb/Session.py]: fixed

File: iotdb-client/client-py/iotdb/Session.py (hunk 4 above)
Comment (HTHou) [reply iotdb-client/client-py/iotdb/Session.py]: fixed

**Review by jt2594838 — APPROVED** [review]

**Pull request merged**